### PR TITLE
test: add regression test for Actor.exit() deadlock from event listeners

### PR DESCRIPTION
**Pull request by vdusek**
`Actor.exit()` / `Actor.fail()` called from within an event listener used to deadlock into a `RecursionError`, since the cleanup path waited on the very listener task that called it.

Fixed upstream in crawlee's `EventManager`: [apify/crawlee-python#2088](https://github.com/apify/crawlee-python/pull/2088) (merged). On Python 3.11 it still deadlocks, because `asyncio.wait_for` there wraps the awaited coroutine in a separate task, which defeats crawlee's self-wait detection. This is a minor edge case, so we're not adding an SDK-side workaround just for Python 3.11.

This PR:
- Adds a regression test for `Actor.exit()` called from an `ABORTING` listener, skipped on Python 3.11.
- Bumps `uv.lock` (not the declared `pyproject.toml` constraint) to `crawlee==1.9.1b4`, so CI exercises the fix.

Follow-up: bump the declared `crawlee` constraint in `pyproject.toml` once a stable release with the fix ships.

*✍️ Drafted by Claude Code*

### CODE REVIEW

File: src/apify/_actor.py [hunk 1]
```diff
@@ -1488,6 +1492,28 @@ def _get_remaining_time(self) -> timedelta | None:
         )
         return None
 
+    @contextmanager
```
Comment (Pijukatel) [inline src/apify/_actor.py]: Taking something out for a while and then returning it back feels like we should not have taken it out in the first place.  This kind of action seems very suspicious in async code, and I am worried about some potential race conditions.
 
Maybe take one step back and try to rethink this instead of fixing this one scenario.

Comment (vdusek) [reply src/apify/_actor.py]: Good point, maybe we can resolve this on the `crawlee-python` level

**Comment by vdusek** [comment]
maybe we can close this in favor of https://github.com/apify/crawlee-python/pull/2088 , let's see

**Comment by vdusek** [comment]
Merge it with Crawlee 1.9.1 in the lockfile

**Review by Mantisus — APPROVED** [review]
LGTM

**Review by B4nan — APPROVED** [review]
